R/describe_distribution.R

```diff
@@ -34,14 +38,79 @@
 #' data(iris)
 #' describe_distribution(iris)
 #' describe_distribution(iris, include_factors = TRUE, quartiles = TRUE)
+#' describe_distribution(list(mtcars$mpg, mtcars$cyl))
 #' @export
 describe_distribution <- function(x, ...) {
   UseMethod("describe_distribution")
 }
 
+#' @rdname describe_distribution
+#' @export
+
+describe_distribution.list <- function(x,
+                                       centrality = "mean",
+                                       dispersion = TRUE,
+                                       iqr = TRUE,
+                                       range = TRUE,
+                                       quartiles = FALSE,
+                                       ci = NULL,
+                                       include_factors = FALSE,
+                                       iterations = 100,
+                                       threshold = .1,
+                                       verbose = TRUE,
+                                       ...) {
+
+  # get elements names as is
+  # ex: list(mtcars$mpg, mtcars$cyl) -> c("mtcars$mpg", "mtcars$cyl")
+  nm <- sapply(sys.call()[[2]], deparse)[-1]
+
+  # Not possible to obtain elements names if they are stored in
+  # an object
+  if (length(nm) == 0) {
+    nm <- paste0("Var_", 1:length(x))
+  }
+
+  out <- do.call(rbind, lapply(x, function(i) {
+    if ((include_factors && is.factor(i)) || (!is.character(i) && !is.factor(i))) {
+      describe_distribution(
+        i,
+        centrality = centrality,
+        dispersion = dispersion,
+        iqr = iqr,
+        range = range,
+        quartiles = quartiles,
+        ci = ci,
+        iterations = iterations,
+        threshold = threshold,
+        verbose = verbose
+      )
+    }
+  }))
+
+
+  if (!is.null(names(x))) {
+    empty_names <- which(names(x) == "")
+    new_names <- names(x)
+    new_names[empty_names] <- nm[empty_names]
+  } else {
+    new_names <- nm
+  }
+
+  out$Variable <- new_names
+  row.names(out) <- NULL
+  out <- out[c("Variable", setdiff(colnames(out), "Variable"))]
+
+  class(out) <- unique(c("parameters_distribution", "see_parameters_distribution", class(out)))
```

Comment:
Have you tried the plotting method in `{see}` and checked if it works with outputs from `list` input?

Maybe you can post an example in the PR thread.

Reply:
I didn't know there was a plotting method for that. I just tried `plot(describe_distribution(list(mtcars$mpg, mtcars$cyl)))` and it doesn't work. I don't have any experience with `ggplot2` programming so not sure how to handle this. I'll take a look at `see` source code later, maybe it's easier than I think

Reply:
I think you can leave this in for now and create an issue in `{see}` repository so that we don't forget about it.